lib/class-wp-theme-json-gutenberg.php

```diff
@@ -714,6 +714,23 @@ private function get_block_classes( $style_nodes ) {
 				}
 			}
 
+			// Set a default margin on the root selector before generating the ruleset
+			// from the `theme.json`. This is to ensure that if the `theme.json` declares
+			// `margin` in its `spacing` declaration for the `body` element then these
+			// user-generated values take precedence in the CSS cascade.
+			// See: https://github.com/WordPress/gutenberg/issues/36147.
+			if ( self::ROOT_BLOCK_SELECTOR === $selector ) {
```

Comment:
I think the comment should be more about why we are zeroing the margin (default browsers margin) and not about the specific issue.

Reply:
Good idea! I had no idea why we were zeroing the margin. I assumed it was a reset or normalization but thanks for clarifying.

I'll update the comment. I would like to retain the explanation about the ordering as it's important to avoid a regression.

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/36960/commits/f7ed1177cbf4034eeb03ec7fdeb60a968b06cbce